composer/core/state.py

```diff
@@ -134,12 +134,13 @@ def __init__(
             else:
                 train_dataloader = DataSpec(train_dataloader)
         self.train_data = train_dataloader
-        if not isinstance(eval_dataloader, DataSpec):
-            if isinstance(eval_dataloader, dict):
-                eval_dataloader = DataSpec(**eval_dataloader)
-            else:
-                eval_dataloader = DataSpec(eval_dataloader)
-        self.eval_data = eval_dataloader
+        for evaluator in evaluators:
+            if not isinstance(evaluator.dataset, DataSpec):
+                if isinstance(evaluator.dataset, dict):
```

Comment:
Can you move this logic to `Evaluator.__init__`, rather than here? (That way it also won't need to be duplicated on the setter below)